File: Front/Timer/src/app/model/Intento.ts
import { Sesion } from './Sesion';
import { Categoria } from './Categoria';

export class Intento {
    id: number;
    fecha: string;
    tiempo: number;
    sesion: Sesion;
    categoria: Categoria;

    constructor(
        id: number = 0,
        fecha: string = '',
        tiempo: number = 0,
        sesion: Sesion = new Sesion(),
        categoria: Categoria = new Categoria()
    ) {
        this.id = id;
        this.fecha = fecha;
        this.tiempo = tiempo;
        this.sesion = sesion;
        this.categoria = categoria;
    }
}

File: apiTimer/src/main/java/com/apiTimer/entidades/Intento.java
package com.apiTimer.entidades;

import com.fasterxml.jackson.annotation.JsonBackReference;
import jakarta.persistence.*;
import java.io.Serializable;
import java.time.LocalDateTime;

@Entity
@Table(name = "Intentos")
public class Intento implements Serializable {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(nullable = false)
    private LocalDateTime fecha;

    @Column(nullable = false)
    private int tiempo; // en milisegundos

    @ManyToOne
    @JoinColumn(name = "sesion_id", nullable = false)
    @JsonBackReference
    private Sesion sesion;

    @ManyToOne
    @JoinColumn(name = "categoria_id", nullable = false)
    private Categoria categoria;

    // Getters and Setters
    public Long getId() {
        return id;
    }

    public void setId(Long id) {
        this.id = id;
    }

    public LocalDateTime getFecha() {
        return fecha;
    }

    public void setFecha(LocalDateTime fecha) {
        this.fecha = fecha;
    }

    public int getTiempo() {
        return tiempo;
    }

    public void setTiempo(int tiempo) {
        this.tiempo = tiempo;
    }

    public Sesion getSesion() {
        return sesion;
    }

    public void setSesion(Sesion sesion) {
        this.sesion = sesion;
    }

    public Categoria getCategoria() {
        return categoria;
    }

    public void setCategoria(Categoria categoria) {
        this.categoria = categoria;
    }
}
